lib/sla.js

```diff
@@ -1,6 +1,77 @@
+/*jslint node: true */
 'use strict';
 
-module.exports = function(callback){
-    console.log("\"snarflegluff\" should be printed below:");
-    var str = "snarflegluff";
-    return str;}
+/**
+ * Receives initial input and begins the application.
+ * @param {Object} client - Client received from app.js.
+ * @callback end - Handles errors and ends the application.
+ */
+module.exports = function(client, end) {
+  designateMixingBridge();
+
+  /**
+   * Looks for an existing mixing bridge and creates one if none exist.
+   */
+  function designateMixingBridge() {
+    client.bridges.list(function(err, bridges) {
+      var bridge = bridges.filter(function(candidate) {
+        return candidate.bridge_type === 'mixing';
+      })[0];
+      if (bridge) {
+        console.log('Using existing mixing bridge %s', bridge.id);
+      } else {
+        bridge = client.bridges.create({type: 'mixing'}, function(err, bridge) {
```

Comment:
This is an asynchronous call, you can't assign the result of it to bridge. The bridge object that is passed into the callback is the object you are looking for.